v2/internal/util/azuresql/azuresql.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package azuresql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	_ "github.com/denisenkom/go-mssqldb"
+	mssql "github.com/denisenkom/go-mssqldb"
+	"github.com/pkg/errors"
+)
+
+// ServerPort is the default server port for sql server
+const ServerPort = 1433
+
+// driverName is driver name for db connection
+const driverName = "sqlserver"
+
+func ConnectToDB(
+	ctx context.Context,
+	serverAddress string,
+	database string,
+	port int,
+	user string,
+	password string,
+) (*sql.DB, error) {
+	// Make sure to set connection timeout: https://github.com/denisenkom/go-mssqldb/issues/609
+	connString := fmt.Sprintf(
+		"server=%s;database=%s;user id=%s;password=%s;port=%d;Persist Security Info=False;Pooling=False;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30",
+		serverAddress,
+		database,
+		user,
+		password,
+		port)
+
+	db, err := sql.Open(driverName, connString)
+	if err != nil {
+		return db, err
+	}
+
+	db.SetConnMaxLifetime(1 * time.Minute)
+	db.SetMaxOpenConns(1)
+	db.SetMaxIdleConns(1)
+
+	err = db.PingContext(ctx)
+	if err != nil {
+		return db, err
+	}
+
+	return db, err
+}
+
+// ConnectToDBAAD connects to the Azure SQL database using the specified token provider.
+// TODO: Does this work for any form of user? Don't need a username at all?
+func ConnectToDBAAD(
```

Comment:
My eye keeps parsing this as `D-BAAD` ... suggest renaming this method.

```suggestion
func ConnectToDBUsingAAD(
```

Reply:
Done